### PR TITLE
Add user report to data dictionary

### PR DESCRIPTION
**Pull request by JCantu248**
#  #

## 🗣 Description ##

This pull request is to merge the changes involving adding the user_report document to the NCATs data dictionary. 

## 💭 Motivation and context ##

With the user_report collection being added to the PCA database, it is necessary to update the data dictionary to describe the new document schema. The changes describe the fields found in the document schema, and their intended purpose. 

This PR should only be merged after https://github.com/cisagov/pca-core/pull/5 is merged 
## 🧪 Testing ##

Changes were made to the NCATS_data_dictionary.md file and previewed, and that the link on the table of contents redirects as expected.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

File: NCATS_Data_Dictionary.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ This information is organized by database and collection (table).
 * [emails Collection](#emails-collection)
 * [templates Collection](#templates-collection)
 * [users Collection](#users-collection)
+* [user_reports Collection](#user_reports-collection)
```
Comment (dav3r) [inline NCATS_Data_Dictionary.md:52]: Keep alphabetical order intact:
```suggestion
* [user_reports Collection](#user_reports-collection)
* [users Collection](#users-collection)
```

File: NCATS_Data_Dictionary.md [hunk 2]
```diff
@@ -741,6 +742,13 @@ sent to the target
 * `customer_defined_labels` [dictionary]: Customer-defined text labels for
 grouping users for statistical purposes
 
+### user_reports Collection ###
+* `customer` [string]: PCA customer identifier that this user_report is associated with
+* `assessment` [string]: PCA assessment identifier that this user_report is associated with
+* `campaign` [string]: PCA campaign identifier that this user_report is associated with
+* `first_report` [datetime]: Datetime for when the first user click is reported for a campaign
+* `total_num_reports` [integer]: The total number of clicks reported for a campaign
```
Comment (dav3r) [inline NCATS_Data_Dictionary.md:750]: Move this `### user_reports Collection ###` block above the `### users Collection ###` block and alphabetize the fields within it.

**Review by dav3r — CHANGES_REQUESTED** [review]
I noted a couple of items related to alphabetical ordering.

File: NCATS_Data_Dictionary.md (hunk 1 above)
Comment (JCantu248) [reply NCATS_Data_Dictionary.md]: Taken care of in commit 0277a31978eace2b882e6981e9ec343e34574209

File: NCATS_Data_Dictionary.md (hunk 2 above)
Comment (JCantu248) [reply NCATS_Data_Dictionary.md]: Taken care of in commit 0277a31978eace2b882e6981e9ec343e34574209

**Review by dav3r — CHANGES_REQUESTED** [review]
I have one other question and a small change request.

File: NCATS_Data_Dictionary.md [hunk 3]
```diff
@@ -734,13 +735,20 @@ sent to the target
   * 0: Does not evoke a sense of greed
   * 1: Email appeals to greed and monetary gain are within the theme of the email
 
+### user_reports Collection ###
+* `assessment` [string]: PCA assessment identifier that this user_report is associated with
+* `campaign` [string]: PCA campaign identifier that this user_report is associated with  
+* `customer` [string]: PCA customer identifier that this user_report is associated with
+* `first_report` [datetime]: Datetime for when the first user click is reported for a campaign
```
[the remark this answers is not in the document]
Comment (JCantu248) [reply NCATS_Data_Dictionary.md:742]: Yes, it is following the same date format. Committing suggestion.

File: NCATS_Data_Dictionary.md [hunk 4]
```diff
@@ -734,13 +735,20 @@ sent to the target
...
+* `total_num_reports` [integer]: The total number of clicks reported for a campaign
+
 ### users Collection ###
 
 * `_id` [ObjectId]: PCA ID of this user
 * `customer` [string]: PCA customer identifier that this user is associated with
 * `customer_defined_labels` [dictionary]: Customer-defined text labels for
 grouping users for statistical purposes
-
+
```
Comment (dav3r) [inline NCATS_Data_Dictionary.md:751]: What happened here?  Looks like some stray whitespace that should be deleted- this should just be an empty line.

**Review by jsf9k — APPROVED** [review]
This looks great to me, once you fix the whitespace issue that @dav3r pointed out.

File: NCATS_Data_Dictionary.md (hunk 4 above)
Comment (JCantu248) [reply NCATS_Data_Dictionary.md]: Take a look at 9e4946ccfe13fba6f361af63b58caf4ab06244c8, I swatted that stray whitespace

Comment (dav3r) [reply NCATS_Data_Dictionary.md]: Thanks!

**Review by dav3r — APPROVED** [review]
Looks great now, though we should not merge this PR until https://github.com/cisagov/pca-core/pull/5 is merged.  @JCantu248 - please mention that dependency somewhere in the `Motivation and context` section of this PR.

**Pull request merged**